doc/nrf-bm/migration/nrf5_bm_migration.rst

```diff
@@ -0,0 +1,231 @@
+
+Build system
+============
+
+In |BMshort|, projects can be organized more freely and managed more easily thanks to the new build system based on CMake and west.
+An application can reside in any folder, and includes a :file:`CMakeLists.txt` build system configuration file (for CMake) and a :file:`prj.conf` project configuration file (for Kconfig, which is a tool that is part of the build system).
+
+The purpose of the :file:`CMakeLists.txt` file is to define the structure of the project, such as source files, libraries, dependencies, and build targets.
+Specifically, this file is used to:
+
+* Define the name of your project.
+* Add source files:
+
+  * Specify the source files for your application that need to be compiled.
+    This includes main files, libraries, or any additional ``.c`` or ``.cpp`` code.
+
+    .. code-block:: cmake
+
+       target_sources(app PRIVATE src/main.c src/other_module.c)
+
+* Include headers:
+
+  * Add directories for custom include paths to ensure header files are correctly found.
+
+    .. code-block:: cmake
+
+       target_include_directories(app PRIVATE include)
+
+* Link libraries:
+
+  * Add external or custom libraries required for your project (such as math, custom vendor libraries).
+
+    .. code-block:: cmake
+
+       target_link_libraries(app PRIVATE libfoo)
+
+* External modules and definitions:
+
+  * If your project relies on external modules or additional functionality, include them.
+
+    .. code-block:: cmake
+
+       add_subdirectory("path_to_directory")
+
+Dependencies are added to the project configuration and incorporated into the build process automatically (you do not have to manually add files from specific locations), with the aid of a tool called Kconfig, in conjunction with CMake.
+That is done using the :file:`prj.conf` file that replaces the :file:`sdk_config.h` file from the nRF5 SDK.
+The entries in the :file:`prj.conf` file are referred to as **Kconfig options**.
+
+Unlike the :file:`sdk_config.h` file that lists all the configuration options relevant for an application, even those whose values are unchanged from defaults, the :file:`prj.conf` file only contains entries whose values are manually set or overridden.
+
+The configuration options whose values are left as default are not present in that file, although when a project is built, a file containing all configuration options pertaining to the application (called :file:`autoconf.h`) is created in the background.
+The build system and the |nRFVSC| extension both provide a way to conveniently browse and search all available project options and inspect their dependencies and read their help text (menuconfig/extension).
+
+There is no distinction or taxonomy between Kconfig options that are applicable to regular |NCS| only and those that are only applicable to |BMlong|.
+Kconfig options that are applicable to the current application are shown and are selectable, while others are not.
```

Comment:
Note/warning?

Reply:
I would leave as is, let's go back to this later.